### PR TITLE
Windows: Prevent platform thread from deadlocking during window resize

### PR DESCRIPTION
**Pull request by abnemo**
Fixes [flutter/flutter#77034](https://github.com/flutter/flutter/issues/77034)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).

### CODE REVIEW

**Comment by jnschulze** [comment]
I was looking at the same issue a couple of days ago and assumed that it would only occur if `FlutterWindowsView::OnWindowSizeChanged` gets invoked before the first frame gets rendered (there was a similar issue on macOS: https://github.com/flutter/engine/pull/22457).

However, I realized that it AFAICT only occurs when attempting to resize the window programmatically using `SetWindowPos` or `MoveWindow`.

I haven't found the root cause yet but I think having a timeout for the resize condition variable is a viable safety net.
It's definitely better than risking to block the platform thread forever.

File: shell/platform/windows/flutter_windows_view.cc [hunk 1]
```diff
@@ -13,23 +13,29 @@
 
 namespace flutter {
 
+namespace {
+// The maximum duration to block the platform thread for while waiting
+// for a window resize operation to complete.
+constexpr std::chrono::milliseconds kWindowResizeTimeout{100};
```
[the remark this answers is not in the document]
Comment (abnemo) [reply shell/platform/windows/flutter_windows_view.cc:19]: I have measured in debug mode how long this method blocks in normal case when manually resizing the window. It took round about 40ms but never more than 60ms.

**Comment by jnschulze** [comment]
I just tested it with the example of the linked issue (https://github.com/flutter/flutter/issues/77034) and can confirm that the window gets properly moved and resized without freezing the app.

**Review by jnschulze — APPROVED** [review]

**Comment by tilongzs** [comment]
Great, thanks a lot for the fix.

**Review by cbracken — COMMENTED** [review]
Nice -- Thanks!

**Pull request merged**